src/dvclive/live.py

```diff
@@ -265,11 +265,18 @@ def _init_dvc(self):  # noqa: C901
             self._include_untracked.append(self.dir)
 
     def _init_dvc_file(self) -> str:
-        if isinstance(self._dvcyaml, str):
-            if os.path.basename(self._dvcyaml) == "dvc.yaml":
-                return self._dvcyaml
-            raise InvalidDvcyamlError
-        return "dvc.yaml"
+        if self._dvcyaml is None:
+            return "dvc.yaml"
+        if isinstance(self._dvcyaml, bool):
+            return "dvc.yaml"
+        if isinstance(self._dvcyaml, Path):
+            self._dvcyaml = str(self._dvcyaml)
```

Comment:
Hi, I'll suggest avoiding `isinstance(..., Path)` checks as much as possible.

You can use `os.fspath()` to convert `Path` or `PathLike` to `str` (`str` are  returned as-is).

Reply:
@skshetry replaced these lines with `os.fspath()`